authentication-pre-request.js

```diff
@@ -0,0 +1,38 @@
+var { Header } = require('postman-collection')
+
+const oauth2Request = {
+    url: 'https://api.oregonstate.edu/oauth2/token',
+    method: 'POST',
+    header: 'Content-Type:application/x-www-form-urlencoded',
+    body:`grant_type=client_credentials&scope=
+      &client_id=${pm.collectionVariables.get('clientID')}
+      &client_secret=${pm.collectionVariables.get('clientSecret')}`
+};
+
+if (pm.environment.name === "LOCAL") {
+    const token = btoa(
+      `${pm.collectionVariables.get('username')}:${pm.collectionVariables.get('password')}`);
+    pm.request.headers.add(new Header("Authorization: Basic " + token));
+} else if (pm.collectionVariables.get('bearerToken')
+            && pm.collectionVariables.get('tokenExpirationDate')
+            && pm.collectionVariables.get('tokenExpirationDate') > new Date().getTime()) {
```

Comment:
`pm.collectionVariables.get('bearerToken')` and `pm.collectionVariables.get('tokenExpirationDate')` are used twice. It should be stored as reusalbe variables.